plugins/search-backend-node/src/engines/ElasticSearchSearchEngine.ts

```diff
@@ -0,0 +1,296 @@
+
+  static async initialize({
+    logger,
+    config,
+    aliasPostfix = `search`,
+    indexPrefix = ``,
+  }: ElasticSearchOptions) {
+    return new ElasticSearchSearchEngine(
+      await ElasticSearchSearchEngine.constructElasticSearchClient(
+        logger,
+        config.getConfig('search.elasticSearch'),
+      ),
+      aliasPostfix,
+      indexPrefix,
+      logger,
+    );
+  }
+
+  private static async constructElasticSearchClient(
+    logger: Logger,
+    config?: Config,
+  ) {
+    if (!config) {
+      throw new Error('No elastic search config found');
+    }
+
+    if (config.getOptionalString('provider') === 'elastic') {
+      logger.info('Initializing Elastic.co ElasticSearch search engine.');
+      return new Client({
+        cloud: {
+          id: config.getString('cloudId'),
+        },
+        auth: config.get<ElasticConfigAuth>('auth'),
+      });
+    }
+    if (config.getOptionalString('provider') === 'aws') {
+      logger.info('Initializing AWS ElasticSearch search engine.');
+      const awsCredentials = await awsGetCredentials();
+      const AWSConnection = createAWSConnection(awsCredentials);
+      return new Client({
+        node: config.getString('node'),
+        ...AWSConnection,
+      });
+    }
+    logger.info('Initializing ElasticSearch search engine.');
+    return new Client({
+      node: config.getString('node'),
+      auth: config.get<ElasticConfigAuth>('auth'),
+    });
+  }
+
+  protected translator({
+    term,
+    filters = {},
+    types,
+  }: SearchQuery): ConcreteElasticSearchQuery {
+    let searchableFields = [
+      ...new Set(
+        types
+          ? types.flatMap(it =>
+              !this.documentTypeFields[it] ? [] : this.documentTypeFields[it],
+            )
+          : Object.values(this.documentTypeFields).reduce(
+              (acc, it) => acc.concat(it),
+              [],
+            ),
+      ),
+    ];
+
+    if (searchableFields.length === 0) {
+      searchableFields = ['*'];
+    }
+
+    return {
+      elasticQueryBuilder: () => {
+        const filter = Object.entries(filters)
+          .filter(([_, value]) => Boolean(value))
+          .map(([key, value]: [key: string, value: any]) => {
+            if (['string', 'number', 'boolean'].includes(typeof value)) {
+              return esb.matchQuery(key, value.toString());
+            }
+            if (Array.isArray(value)) {
+              return esb
+                .boolQuery()
+                .should(value.map(it => esb.matchQuery(key, it.toString())));
+            }
+            this.logger.error(
+              'Failed to query, unrecognized filter type',
+              key,
+              value,
+            );
+            throw new Error(
+              'Failed to add filters to query. Unrecognized filter type',
+            );
+          });
+        const query = isBlank(term)
+          ? esb.matchAllQuery()
+          : esb
+              .multiMatchQuery(searchableFields, term)
```

Comment:
Opting to search across all fields with a ElasticSearch "fuzzy" search in this initial implementation. This query is the most likely one that needs to be tweaked when we know more about the datasets and better how we want query them.